Library/Homebrew/resource.rb

```diff
@@ -154,12 +156,19 @@ def mirror(val)
 
   private
 
+  # TODO fix and update for HeadVersion, remove HeadVersion at all if needed
+  # as it might be unreachable at all
   def detect_version(val)
     return if val.nil? && url.nil?
 
     case val
     when nil     then Version.detect(url, specs)
-    when String  then Version.new(val)
+    when String
+      if val.start_with?("HEAD")
+        HeadVersion.new(val)
+      else
+        Version.new(val)
+      end
```

Comment:
Should this be using `Version.parse_version` to avoid the slight code duplication?